src/averageAges.js

```diff
@@ -55,9 +70,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(
+    person => people.some(
+      mother => person.mother === mother.name)
+      && (!onlyWithSon || person.sex === 'm'));
+
+  const suma = children.map(
+    child => (child.born - people.find(
+      mother => mother.name === child.mother).born)).reduce(
+    (a, b) => (a + b), 0);
+
+  return average(suma, children);
 }
 
+function average(sum, people) {
+  return Math.round(sum / people.length * 100) / 100;
```

Comment:
not sure why are you multiplying and then dividing by 100 here. could you explain?